src/routes/safe/components/Settings/index.tsx

```diff
@@ -75,89 +65,35 @@ const Settings: React.FC = () => {
-            <Row className={cn(classes.menuOption, menuOptionIndex === 3 && classes.active)} onClick={handleChange(3)}>
-              <IconText
-                iconSize="sm"
-                textSize="xl"
-                iconType="requiredConfirmations"
-                text="Policies"
-                color={menuOptionIndex === 3 ? 'primary' : 'secondary'}
-              />
-            </Row>
-            <Hairline className={classes.hairline} />
-            {isSpendingLimitEnabled && (
-              <>
-                <Row
-                  className={cn(classes.menuOption, menuOptionIndex === 4 && classes.active)}
-                  onClick={handleChange(4)}
-                >
-                  <IconText
-                    iconSize="sm"
-                    textSize="xl"
-                    iconType="fuelIndicator"
-                    text="Spending limit"
-                    color={menuOptionIndex === 4 ? 'primary' : 'secondary'}
-                  />
-                </Row>
-                <Hairline className={classes.hairline} />
-              </>
-            )}
-            <Row className={cn(classes.menuOption, menuOptionIndex === 5 && classes.active)} onClick={handleChange(5)}>
-              <IconText
-                iconSize="sm"
-                textSize="xl"
-                iconType="settingsTool"
-                text="Advanced"
-                color={menuOptionIndex === 5 ? 'primary' : 'secondary'}
-              />
-            </Row>
-            <Hairline className={classes.hairline} />
-          </Block>
-        </Col>
         <Col className={classes.contents} layout="column">
           <Block className={classes.container}>
-            {menuOptionIndex === 1 && <SafeDetails />}
-            {menuOptionIndex === 2 && <ManageOwners granted={granted} owners={owners} />}
-            {menuOptionIndex === 3 && <ThresholdSettings />}
-            {menuOptionIndex === 4 && <SpendingLimitSettings />}
-            {menuOptionIndex === 5 && <Advanced />}
+            <Switch>
+              <Route
+                path={`${SAFELIST_ADDRESS}/${address}/settings/details`}
+                exact
+                render={() => <SafeDetails />}
+              ></Route>
+              <Route
+                path={`${SAFELIST_ADDRESS}/${address}/settings/owners`}
+                exact
+                render={() => <ManageOwners granted={granted} owners={owners} />}
+              ></Route>
+              <Route
+                path={`${SAFELIST_ADDRESS}/${address}/settings/policies`}
+                exact
+                render={() => <ThresholdSettings />}
+              ></Route>
+              <Route
+                path={`${SAFELIST_ADDRESS}/${address}/settings/spending-limit`}
+                exact
+                render={() => <SpendingLimitSettings />}
+              ></Route>
+              <Route
+                path={`${SAFELIST_ADDRESS}/${address}/settings/advanced`}
```

Comment:
These URLs are duplicated here and in the menu. Ideally they should come from the same enum, and the menu should use `generatePath`.

Reply:
With current implementation I can only do a partial refactor for this suggestion. I won't be able to use `generatePath`, as SAFELIST_ADDRESS include a `/` character and generatePath escape those values.

To improve this we should either go for a tech debt ticket because it won't make sense to change it only here or to define two  `SAFELIST_ADDRESS` values in `src/routes/routes` meanwhile we can't do the full migration